### ISSUE TITLE
Release.Namespace and Release.Time in charts repo but missing from Helm v3

### ISSUE THREAD

**Issue by mattfarina** [helm/helm#5732]
The dev-v3 branch, as of c62a3a4896e7f6af57c7282c4415f523195523d4, no longer has `.Release.Namespace` and `.Release.Time` available in templates. It appears this was changed in https://github.com/helm/helm/commit/aa859e3f88291c94220d26e9d1127c1557c2af98 in order to "make template rendering discrete and repeatable".

There are some problems with this.

1. Charts in the charts repo use both of these features. See [`.Release.Time` here](https://github.com/helm/charts/search?q=Release.Time&unscoped_q=Release.Time) and [`.Release.Namespace` here](https://github.com/helm/charts/search?q=Release.Namespace&unscoped_q=Release.Namespace). These charts won't work with Helm v3
2. `.Release.Namespace` is needed for RBAC. Removing it hinders the ability of a chart to create RBAC config. See the comment on the commit and how it's used in the charts repo. Removal of namespace limits what charts can be used for

This was referenced in https://github.com/helm/helm/issues/4255#issuecomment-399491963 where @adamreese said it would be added back.

cc @bacongobbler @adamreese

**Comment by technosophos** [comment]
So `.Release.Time` is effectively the same as calling `now` as a template function. And when I looked at the use cases, most might be better served by calling a random function instead. I am in favor of not putting that one back in if we can get away with it.

`.Release.Namespace` got merged back in on #5737 (as a result of this issue). So basically, if we can decide on `.Release.Time`, that's all that is left for this issue.

**Comment by mattfarina** [comment]
There aren't many public  uses of `.Release.Time` from what I can tell. I would suggest:

1. We update the cases in the charts repo to use the template function instead
2. We document the use of the template function instead as part of the 2->3 docs

Once we do those two things we can close this out unless I missed something.

**Comment by mattfarina** [comment]
@technosophos So... everyone is using `.Release.Time.Seconds` which is seconds in UTC since the unix epoch. There is no golang format pattern to get this. So, we need a different way to handle this.

If we add a function to sprig there will be many using a version of Helm that doesn't support this. Sadly, lots of people use old helm versions. Not sure if they are using this part of release.

I created https://github.com/Masterminds/sprig/pull/164 to give sprig the ability to do this.

**Comment by bacongobbler** [comment]
Are there action items we should be addressing for Helm 3, or is this safe to close with the sprig bump?

**Comment by bacongobbler** [comment]
perhaps this should be closed in favour of #5733 if that's the last action item to close this one out.

**Comment by bacongobbler** [comment]
closing in favour of #5733.

**Comment by KIVagant** [comment]
> So .Release.Time is effectively the same as calling now as a template function.

@technosophos, but isn't there a risk that `now` can return 1 second difference if it is called in several templates when Helm is processing them very close to a second end? This tiny difference could result in a deployment that could not be able to find it's pod because a label is +/-1 second different?

Was not the `.Release.Time` calling `time.Now()` only once and then every subsequent call would return 100% the same number?

**Comment by KIVagant** [comment]
Also, is there a way of getting a consistent random value that would work equally through all templates (without writing a new helper and defining variables in global scope)?
